ouroboros-consensus/src/Ouroboros/Consensus/Fragment/InFuture.hs

```diff
@@ -47,10 +45,9 @@ import           Ouroboros.Consensus.Util.Time
 
 data CheckInFuture m blk = CheckInFuture {
        -- | POSTCONDITION:
-       --
-       -- > checkInFuture vf >>= \(af, fut) ->
-       -- >   validatedFragment vf == af <=> null fut
-       checkInFuture :: ValidatedFragment (Header blk) (LedgerState blk)
+       -- > checkInFuture _ af >>= \(af', fut) ->
+       -- >   af == af' <=> null fut
+       checkInFuture :: LedgerState blk -> AnchoredFragment (Header blk)
```

Comment:
Please add a comment explaining the assumptions about the given ledger state and the given fragment.